File: examples/python/01_two_arm_bounded.py
"""Simulated two-arm bounded-outcome CMR example."""

import numpy as np

import cmrdesign as cmr

rng = np.random.default_rng(101)
n_pilot = 800
d = np.r_[np.ones(n_pilot // 2), np.zeros(n_pilot // 2)]
y = np.r_[rng.beta(2, 5, n_pilot // 2), rng.beta(4, 4, n_pilot // 2)]

fit = cmr.cmr_two_arm(y, d, alpha=0.05, method="bounded")

print({"pi_treatment": fit.pi, "certificate": fit.U_CMR})

File: examples/r/01_two_arm_bounded.R
# Simulated two-arm bounded-outcome CMR example.

library(cmrdesign)

set.seed(101)
n_pilot <- 800L
d <- c(rep(1, n_pilot / 2), rep(0, n_pilot / 2))
y <- c(rbeta(n_pilot / 2, 2, 5), rbeta(n_pilot / 2, 4, 4))

fit <- cmr_two_arm(y, d, alpha = 0.05, method = "bounded")
allocation <- realize_allocation(fit, n_main = 1000)

print(list(pi_treatment = fit$pi, certificate = fit$U_CMR))
print(list(counts = allocation$counts,
           realized_certificate = allocation$realized_U_CMR))
